lib/Drush/Boot/DrupalBoot8.php

```diff
@@ -118,15 +120,26 @@ function bootstrap_drupal_database() {
   }
 
   function bootstrap_drupal_configuration() {
+    global $drupal_update;
     $this->request = Request::createFromGlobals();
     $classloader = drush_drupal_load_autoloader(DRUPAL_ROOT);
     // @todo - use Request::create() and then no need to set PHP superglobals
     $kernelClass = new \ReflectionClass('\Drupal\Core\DrupalKernel');
     if ($kernelClass->hasMethod('addServiceModifier')) {
-      $this->kernel = DrupalKernel::createFromRequest($this->request, $classloader, 'prod', DRUPAL_ROOT);
+      if (!$drupal_update) {
+        $this->kernel = DrupalKernel::createFromRequest($this->request, $classloader, 'prod');
+      }
+      else {
+        $this->kernel = UpdateKernel::createFromRequest($this->request, $classloader, 'prod');
+      }
     }
     else {
-      $this->kernel = DrushDrupalKernel::createFromRequest($this->request, $classloader, 'prod', DRUPAL_ROOT);
+      if (!$drupal_update) {
+        $this->kernel = DrushDrupalKernel::createFromRequest($this->request, $classloader, 'prod');
+      }
+      else {
+        $this->kernel = DrushDrupalUpdateKernel::createFromRequest($this->request, $classloader, 'prod');
```

Comment:
Is `Drupal\Core\Update\UpdateKernel` a new addition to Drupal 8, or has it been around for a while? If there isn't a version of Drupal that has an `UpdateKernel` that does not have an `addServiceModifier()` method, then we don't need this adapter. Since Drush has not supported this capability at all until now, I am wondering if we can perhaps save ourselves an extra adapter and just fail here for old versions of Drupal.